app/src/main/java/com/example/firstandroid/presentation/add_bathroom/NewBathroomViewModel.kt

```diff
@@ -21,36 +23,41 @@ class NewBathroomViewModel() : ViewModel() {
         .addConverterFactory(GsonConverterFactory.create())
         .build()
 
-    fun updateTitle(title: String) {
+    fun newTitle(title: String) {
         state = state.copy(title = title)
     }
 
-    fun updateLocation(location: String) {
+    fun newLocation(location: String) {
         state = state.copy(location = location)
     }
 
-    fun updateCapacity(capacity: String) {
+    fun newCapacity(capacity: String) {
         capacity.toIntOrNull()?.let {
             state = state.copy(capacity = it)
         }
     }
 
-    fun updateFree(free: Boolean) {
+    fun isChecked(free: Boolean) {
         state = state.copy(free = free)
     }
 
-    fun updateCost(cost: String) {
+    fun newCost(cost: String) {
         cost.toBigDecimalOrNull()?.let {
             state = state.copy(cost = it)
         }
     }
 
-    fun updateHours(hours: String) {
+    fun newHours(hours: String) {
         state = state.copy(hours = hours)
     }
 
-    suspend fun addBathroom(state: NewBathroomState) {
-        retrofit.create(BathroomRepository::class.java)
-        bathroomRepository.addBathroomData(state)
+    fun onStart() {
+        bathroomRepository = retrofit.create(BathroomRepository::class.java)
+        viewModelScope.launch {
+            addBathroom(state)
+        }
+    }
```

Comment:
I have a feeling this state argument is not being used, and you are just using the state object in the ViewModel.

I think you could update this to take no arguments.